src/expint.jl

```diff
@@ -75,62 +75,70 @@ macro E₁_taylor64(z, n::Integer)
 end
 
 # adapted from Steven G Johnson's initial implementation: issue #19
-function expint_opt(x::Float64)
+function expint_opt(x::Float64, expscaled::Bool=false)
```

Comment:
You can make this a compile-time constant by:
```jl
function expint_opt(x::Float64, ::Val{expscaled}=Val{false}()) where {expscaled}
```
and then passing `Val{true}()` in `expinti`.

Reply:
(This tells the compiler to compile different versions specialized for the cases of `Val{true}()` and `Val{false}()` — Julia's compiler specializes on types, so you can force specialization by moving a value into a type.)

Reply:
See also: https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-value-type